.github/workflows/docsbuild-release.yaml

```
@@ -0,0 +1,78 @@
+name: ansible-release-porting-guide
+on:
+  workflow_dispatch:
+    inputs:
+      ansible-version:
+        description: >-
+          Release Version. Example: 11.1.0
+        required: true
+env:
+  CI_COMMIT_MESSAGE: 'Add Ansible community "${{ inputs.ansible-version }}" porting guide'
+
+jobs:
+  build:
+    name: 'Build Porting-Guide (${{ inputs.ansible-version }})'
+    runs-on: ubuntu-latest
+
+    steps:
+
+      - name: Retrieve the major version component
+        id: compute-version
+        run: |
+          echo -n 'major-version=' >> "${GITHUB_OUTPUT}"
+          echo '${{ inputs.ansible_version }}' | sed 's#\([0-9]\+\)\..*#\1#' >> "${GITHUB_OUTPUT}"
+
+      - name: Check out ansible-documentation
+        uses: actions/checkout@v3
+        with:
+          path: ansible-documentation
+          ref: devel
+
+      - name: Checking out to a new branch & setting the user details
+        working-directory: ansible-documentation
+        run: |
+          git checkout -b "publish-${{ inputs.ansible-version }}"
+          git config --global user.name "Github Actions"
+          git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+
+# Get the Porting guide from the ansible-build-data git repo
+
+      - name: Check out ansible-build-data
+        uses: actions/checkout@v3
+        with:
+          repository: anweshadas/ansible-build-data
+          ref: main
+          path: ansible-build-data
+
+      - name: Commit ansible-documentation
+        run: |
+          cp ansible-build-data/${{ steps.compute-version.outputs.major-version }}/porting_guide_${{ steps.compute-version.outputs.major-version }}.rst ./ansible-documentation/docs/docsite/rst/porting_guides/
```

Let's make the YAML lines shorter, bash more verbose and the step name relevant:
```suggestion
      - name: Copy the porting guide document to ansible-documentation
        run: >-
          cp -v
          'ansible-build-data/${{ steps.compute-version.outputs.major-version }}/porting_guide_${{ steps.compute-version.outputs.major-version }}.rst'
          './ansible-documentation/docs/docsite/rst/porting_guides/'
```